x/mock/simulation/random_simulate_blocks.go

```diff
@@ -123,15 +126,27 @@ func SimulateFromSeed(
 	DisplayEvents(events)
 }
 
+func getKeys(validators map[string]mockValidator) []string {
+	keys := make([]string, len(validators))
+	i := 0
+	for key := range validators {
+		keys[i] = key
+		i++
+	}
+	return keys
+}
+
 // RandomRequestBeginBlock generates a list of signing validators according to the provided list of validators, signing fraction, and evidence fraction
 func RandomRequestBeginBlock(t *testing.T, r *rand.Rand, validators map[string]mockValidator, livenessTransitions TransitionMatrix, evidenceFraction float64,
 	pastTimes []time.Time, event func(string), header abci.Header, log string) abci.RequestBeginBlock {
 	if len(validators) == 0 {
 		return abci.RequestBeginBlock{Header: header}
 	}
 	signingValidators := make([]abci.SigningValidator, len(validators))
-	i := 0
-	for _, mVal := range validators {
+	keys := getKeys(validators)
+
+	for i := 0; i < len(keys); i++ {
```

Comment:
Why not just, `for _, key := range  getKeys(validators) { mVal := validators[key] }`?